finetuner/tuner/evaluation.py

```diff
@@ -0,0 +1,161 @@
+        """
+        Convert a Jina document to a relevance representation
+        :param doc: A Jina document. The matched document identifiers, predicted by the search system, are expected to
+            be under ``doc.matches``. The expected documents identifiers along with the relevance scores, given by the
+            user as matching ground truth, should be under ``doc.tags[finetuner.__default_tag_key__]['targets']``.
+        :return: The relevance vector and the total number of relevant documents.
+        """
+        targets: Dict[str, int] = doc.tags[__default_tag_key__]['targets']
+        return [
+            targets[match.id] if match.id in targets else 0 for match in doc.matches
+        ], len(targets)
+
+    def _parse_eval_docs(self) -> DocumentArray:
+        """
+        Convert the evaluation docs to the internal representation used by the Evaluator
+        """
+        summmary_docs = DocumentArray()
+        for doc in self._eval_data:
+            relevancies = [
+                (m.id, m.tags[__default_tag_key__])
+                for m in doc.matches
+                if m.tags[__default_tag_key__] > 0
+            ]
+            relevancies = sorted(relevancies, key=lambda x: x[1])
+            summmary_doc = Document(
+                id=doc.id,
+                tags={__default_tag_key__: {'targets': dict(relevancies)}},
+            )
+            summmary_docs.append(summmary_doc)
+
+        return summmary_docs
+
+    def _score_docs(self) -> None:
+        """
+        Emded the evaluation docs and compute the matches from the catalog. Evaluation docs
+        embeddings are overwritten, but matches are not
+        """
+        if self._embed_model is not None:
+            embed(self._eval_data, self._embed_model)
+            embed(self._catalog, self._embed_model)
+
+        for doc in self._summary_docs:
+            doc.embedding = self._eval_data[doc.id].embedding
+            doc.matches.clear()
+
+        self._summary_docs.match(
+            self._catalog, limit=self._limit, metric=self._distance
+        )
+
+    def _get_mean_metrics(self, label: str = 'metrics') -> Dict[str, float]:
+        """
+        Compute the mean metric values across the evaluation docs
+        """
+        means = {}
+        for name, _ in METRICS.items():
+            values = [
+                self._eval_data[doc.id].tags[__default_tag_key__][label][name]
+                for doc in self._summary_docs
+            ]
+            means[f"mean_{name}"] = sum(values) / len(values)
+
+        return means
+
+    @classmethod
+    def list_available_metrics(cls) -> List[str]:
+        """
+        List available metrics
+        """
+        return list(METRICS.keys())
+
+    def evaluate(self, label: str = 'metrics') -> Dict[str, float]:
+        """
+        Run evaluation
+        :param label: Per document metrics are written in each evaluation document under
+            ``doc.tags[__default_tag_key__][label]``.
```

Comment:
apply changes to docstring as well

Reply:
this actually holds. Metrics will be written in `doc.tags["finetuner_label"]["metrics"]` when `label = "metrics"`